### PR TITLE
New Top Navigation to Settings page

### PR DESCRIPTION
**Pull request by eiffelwong1**
### Thanks for everyone joining the Sunday Session
- kwesibatchelor
- Thisisdo

## Summary (1-2 sentences)

complimenting @tonydangblog's PR, after logging in user can navigate to the Settings page. 

## Details (reason and description of the changes)

Used Headless UI Menu bar; made reusable component with a basic transition animation. and used in the top navigation bar with Settings page navigation and logout button.

## Screenshots/Recordings (if applicable)

https://github.com/social-plan-it/plan-it-social-web/assets/29219684/47bf9cf9-1c94-4206-9306-016e4f2ab739

### CODE REVIEW

File: app/components/ui/menu.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      }
+    >
+      <>
+        <Menu.Item>
+          <Link to="/settings">Settings</Link>
+        </Menu.Item>
+        <Menu.Item>
+          <LogoutButton />
+        </Menu.Item>
+      </>
```
Comment (tonydangblog) [inline app/components/ui/menu.tsx:23]: Just curious, is this fragment required?

File: app/components/ui/menu.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    </StyledMenu>
+  );
+}
+
+export function StyledMenu({ button, children }: { button: any; children: ReactElement }) {
```
Comment (tonydangblog) [inline app/components/ui/menu.tsx:28]: Don't think this menu is used externally anywhere so maybe we can skip the `export`?

**Review by tonydangblog — APPROVED** [review]
Great job incorporating `headlessui` components to get the sweet transitions :)

File: app/components/ui/menu.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import { Menu, Transition } from '@headlessui/react';
+import { type ReactElement } from 'react';
+import { UserCircleIcon } from '~/components/ui/icons';
+import { Link } from '~/components/ui/links';
+import { LogoutButton } from '~/components/layout/top-nav';
+
+export function TopNavUserMenu() {
```
Comment (andrelandgraf) [inline app/components/ui/menu.tsx:7]: As discussed, let's move this into `top-nav` as it's not a primitive UI component.

File: app/components/ui/menu.tsx (hunk 1 above)
Comment (eiffelwong1) [reply app/components/ui/menu.tsx:23]: yea, as the the children of StyledMenu are expecting a single element

File: app/components/ui/menu.tsx (hunk 2 above)
Comment (eiffelwong1) [reply app/components/ui/menu.tsx:28]: :D now it external

File: app/components/layout/top-nav.tsx [hunk 4]
```diff
@@ -117,3 +125,26 @@ function LogoutButton({ className }: { className?: string }) {
...
+        <>
+          <Menu.Item>
+            <Link to="/settings">Settings</Link>
+          </Menu.Item>
+          <Menu.Item>
+            <LogoutButton />
+          </Menu.Item>
+        </>
+      </StyledMenu>
+    </div>
```
Comment (eiffelwong1) [inline app/components/layout/top-nav.tsx:148]: for some reason, the div is needed, if not it behaves differently where the button gets displaced

File: app/components/layout/top-nav.tsx [hunk 5]
```diff
@@ -117,3 +125,26 @@ function LogoutButton({ className }: { className?: string }) {
     </Form>
   );
 }
+
+export function TopNavUserMenu() {
+  return (
+    <div>
```
Comment (andrelandgraf) [inline app/components/layout/top-nav.tsx:131]: NIT: We could style the menu a bit but can also be a follow-up!

File: app/components/ui/menu.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import { Menu, Transition } from '@headlessui/react';
+import { type ReactElement } from 'react';
+
+export function StyledMenu({ button, children }: { button: any; children: ReactElement }) {
```
Comment (andrelandgraf) [inline app/components/ui/menu.tsx:4]: NIT: we could add this to our Storybook as it's a reusable UI component!

**Review by CamilingJS — APPROVED** [review]
LGTM!

File: app/components/layout/top-nav.tsx [hunk 7]
```diff
@@ -103,7 +111,7 @@ export function TopNav() {
   );
 }
 
-function LogoutButton({ className }: { className?: string }) {
+export function LogoutButton({ className }: { className?: string }) {
```
Comment (tonydangblog) [inline app/components/layout/top-nav.tsx:114]: Doesn't look like this export is used anywhere else in the codebase. Is the plan to reuse it in the mobile view later?

File: app/components/layout/top-nav.tsx (hunk 4 above)
Comment (tonydangblog) [reply app/components/layout/top-nav.tsx:148]: Weird. Maybe we should leave an inline comment so other devs in the future will know to not remove the `div` on accident?

**Review by tonydangblog — APPROVED** [review]
Left a couple comments, but otherwise looks good to ship to me. :)

**Review by andrelandgraf — APPROVED** [review]

**Pull request merged**